CIS 2.6.1 crashes if there is a secret of type kubernetes.io/tls

#### Setup Details
 CIS Version : 2.6.1
 Build: f5networks/k8s-bigip-ctlr:2.6.1
 BIGIP Version: Big IP v16.1.1 (Build 0.0.16)
 AS3 Version: 3.28
 Agent Mode: AS3    
 Orchestration: K8S      
 Orchestration Version: 1.20.12
 Pool Mode: Nodeport      

#### Description  
CIS crashes with the following error if there is a secret deployed to a namespace of type `kubernetes.io/tls`.
Next to that I would like to add that I would expect setting `--use-secrets=false` to not look for any secrets and therefore the CIS container not run into that issue. However, even with that argument CIS crashes with the following error:

Secret yaml
```
apiVersion: v1
data:
  tls.crt: SOMEBASE64
  tls.key: SOMEBASE64
kind: Secret
metadata:
  name: kubecrypt
  namespace: network
type: kubernetes.io/tls

```

Log output of crashing CIS:
```
2021/11/16 15:14:30 [INFO] [CORE] Registered BigIP Metrics
2021/11/16 15:14:30 [DEBUG] [2021-11-16 15:14:30,888 urllib3.connectionpool DEBUG] https://10.123.200.18:443 "POST /mgmt/shared/authn/login HTTP/1.1" 200 724
2021/11/16 15:14:30 [DEBUG] [2021-11-16 15:14:30,890 urllib3.connectionpool DEBUG] Starting new HTTPS connection (1): 10.123.200.18:443
I1116 15:14:32.040113       1 request.go:668] Waited for 1.083309931s due to client-side throttling, not priority and fairness, request: GET:https://10.203.32.1:443/api/v1/namespaces/f5lab/services?limit=500&resourceVersion=0
2021/11/16 15:14:32 [DEBUG] [CORE] Finished syncing namespace network (9.158µs)
E1116 15:14:32.455164       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 202 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x193d240, 0x28a98f0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x193d240, 0x28a98f0)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).getSecretServiceQueueKeyForIngress(0xc0001bcdc0, 0xc00099fe80, 0x1b6182f, 0x7, 0xc00059e620)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/validateResources.go:192 +0x75
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).checkValidSecrets(0xc0001bcdc0, 0x1b3f1a0, 0xc00099fe80, 0xc00071170e, 0x2, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/validateResources.go:277 +0xe5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).enqueueSecrets(0xc0001bcdc0, 0x1b3f1a0, 0xc00099fe80, 0x1b60c62, 0x6)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:827 +0x46
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).newAppInformer.func14(0x1b3f1a0, 0xc00099fe80)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:752 +0x53
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/shared_informer.go:777 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000598760)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00071df60, 0x1d507e0, 0xc0002678c0, 0x18fd101, 0xc000714060)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000598760, 0x3b9aca00, 0x0, 0x30763a3436646d01, 0xc000714060)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000660600)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0002e6d80, 0xc00007e7c0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
2021/11/16 15:14:32 [DEBUG] [CORE] Periodic enqueue of Service from Namespace: n1
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x16d9735]

goroutine 202 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x193d240, 0x28a98f0)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).getSecretServiceQueueKeyForIngress(0xc0001bcdc0, 0xc00099fe80, 0x1b6182f, 0x7, 0xc00059e620)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/validateResources.go:192 +0x75
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).checkValidSecrets(0xc0001bcdc0, 0x1b3f1a0, 0xc00099fe80, 0xc00071170e, 0x2, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/validateResources.go:277 +0xe5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).enqueueSecrets(0xc0001bcdc0, 0x1b3f1a0, 0xc00099fe80, 0x1b60c62, 0x6)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:827 +0x46
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).newAppInformer.func14(0x1b3f1a0, 0xc00099fe80)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:752 +0x53
k8s.io/client-go/tools/cache.ResourceEventHandlerFuncs.OnAdd(...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/controller.go:231
k8s.io/client-go/tools/cache.(*processorListener).run.func1()
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/shared_informer.go:777 +0xc2
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000598760)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc000709f60, 0x1d507e0, 0xc0002678c0, 0x18fd101, 0xc000714060)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000598760, 0x3b9aca00, 0x0, 0x30763a3436646d01, 0xc000714060)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
k8s.io/client-go/tools/cache.(*processorListener).run(0xc000660600)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/client-go/tools/cache/shared_informer.go:771 +0x95
k8s.io/apimachinery/pkg/util/wait.(*Group).Start.func1(0xc0002e6d80, 0xc00007e7c0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:73 +0x51
created by k8s.io/apimachinery/pkg/util/wait.(*Group).Start
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:71 +0x65
```

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -19,6 +19,7 @@ Bug Fixes
 * :issues:`1873` Enable /metrics endpoint with crd mode.
 * :issues:`1659` Report "status" of TransportServer CRD
 * :issues:`2006` Add support for Wildcard domain name with TLSProfile and VirtualServer
+* :issues:`2102,2016` Fixed crash while validating secrets
 
 2.6.1
 -------------
@@ -432,11 +433,11 @@ Added Functionality
 
 Bug Fixes
 `````````
-* :issues:`1420` Enhanced performance for lower BIG-IP CPU Utilization with optimized CCCL calls.
-* :issues:`1362` CIS supports HTTP Header with iv-groups
-* :issues:`1388,1311` CIS properly manages AS3 ConfigMaps when configured with namespace-labels.
-* :issues:`1337` CIS supports multiple AS3 ConfigMaps
-* :issues:`1171` CIS will not create `_AS3` partition anymore
+* :issues: 1420 Enhanced performance for lower BIG-IP CPU Utilization with optimized CCCL calls.
+* :issues: 1362 CIS supports HTTP Header with iv-groups
+* :issues: 1388,1311 CIS properly manages AS3 ConfigMaps when configured with namespace-labels.
+* :issues: 1337 CIS supports multiple AS3 ConfigMaps
+* :issues: 1171 CIS will not create `_AS3` partition anymore
 
 Vulnerability Fixes
 ```````````````````

--- a/pkg/agent/as3/as3Manager.go
+++ b/pkg/agent/as3/as3Manager.go
@@ -46,7 +46,7 @@ const (
 	as3template          = "template"
 	//as3SchemaLatestURL   = "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json"
 	as3defaultRouteDomain = "defaultRouteDomain"
-	as3SchemaFileName = "as3-schema-3.30.0-5-cis.json"
+	as3SchemaFileName     = "as3-schema-3.30.0-5-cis.json"
 )
 
 var baseAS3Config = `{

--- a/pkg/appmanager/validateResources.go
+++ b/pkg/appmanager/validateResources.go
@@ -157,7 +157,10 @@ func (appMgr *Manager) getSecretServiceQueueKeyForConfigMap(secret *v1.Secret) [
 	var keyList []*serviceQueueKey
 	// We will be adding ResourceKind as Configmaps so that particular Configmaps can be re-synced
 	if !appMgr.AgentCIS.IsImplInAgent(ResourceTypeCfgMap) {
-		appInf, _ := appMgr.getNamespaceInformer(secret.ObjectMeta.Namespace)
+		appInf, ok := appMgr.getNamespaceInformer(secret.ObjectMeta.Namespace)
+		if !ok || appInf.cfgMapInformer == nil {
+			return keyList
+		}
 		configmaps := appInf.cfgMapInformer.GetIndexer().List()
 		for _, obj := range configmaps {
 			cm := obj.(*v1.ConfigMap)
@@ -186,7 +189,7 @@ func (appMgr *Manager) getSecretServiceQueueKeyForIngress(secret *v1.Secret) []*
 	var keyList []*serviceQueueKey
 	// We will be adding ResourceKind as Ingress so that particular ingress can be re-synced
 	appInf, ok := appMgr.getNamespaceInformer(secret.ObjectMeta.Namespace)
-	if !ok {
+	if !ok || appInf.ingInformer == nil {
 		return keyList
 	}
 	ingresses := appInf.ingInformer.GetIndexer().List()
@@ -273,13 +276,15 @@ func (appMgr *Manager) checkValidSecrets(
 	if _, ok := secret.Data["tls.key"]; !ok {
 		return false, nil
 	}
-	// Getting the ServiceQueue key for ingresses
-	keyList := appMgr.getSecretServiceQueueKeyForIngress(secret)
-	// appending the ServiceQueue key for configmaps
-	keyList = append(keyList, appMgr.getSecretServiceQueueKeyForConfigMap(secret)...)
+	if appMgr.useSecrets {
+		// Getting the ServiceQueue key for ingresses
+		keyList := appMgr.getSecretServiceQueueKeyForIngress(secret)
+		// appending the ServiceQueue key for configmaps
+		keyList = append(keyList, appMgr.getSecretServiceQueueKeyForConfigMap(secret)...)
 
-	if len(keyList) > 0 {
-		return true, keyList
+		if len(keyList) > 0 {
+			return true, keyList
+		}
 	}
 	// As no Virtual server is using this secret we will skip processing
 	return false, nil

